src/kbmod/search/image_kernels.cu

```diff
@@ -236,9 +236,88 @@ extern "C" void deviceGrowMask(int width, int height, float *source,
     checkCudaErrors(cudaFree(deviceResultImg));
 }
 
+extern "C" __device__ __host__ pixelPos findPeakImageVect(int width, int height, float *img,
+                                                          bool furthest_from_center) {
+    int c_x = width / 2;
+    int c_y = height / 2;
+
+    // Initialize the variables for tracking the peak's location.
+    pixelPos result = { 0, 0 };
+    float max_val = img[0];
+    float dist2 = c_x * c_x + c_y * c_y;
+
+    // Search each pixel for the peak.
+    for (int y = 0; y < height; ++y) {
+        for (int x = 0; x < width; ++x) {
+            if (img[y * width + x] > max_val) {
+                max_val = img[y * width + x];
```

Comment:
The compiler likely is going to keep `img[y * width + x]` in a register for reuse, but it might be good to capture it into a variable just to be sure this doesn't cause multiple global memory reads.

Reply:
Good call. I fixed it.